client/me/purchases/confirm-cancel-domain/index.jsx

```diff
@@ -0,0 +1,252 @@
+
+				return;
+			}
+
+			notices.success(
+				this.translate( '%(purchaseName)s was successfully cancelled and refunded.', {
+					args: { purchaseName }
+				} ), { persistent: true }
+			);
+
+			clearPurchases();
+
+			analytics.tracks.recordEvent(
+				'calypso_domain_cancel_form_submit',
+				{ product_slug: purchase.productSlug }
+			);
+
+			page.redirect( paths.list( this.props.selectedSite.slug ) );
+		} );
+	},
+
+	onReasonChange( newReason ) {
+		this.setState( { selectedReason: newReason } );
+	},
+
+	onConfirmationChange() {
+		this.setState( { confirmed: ! this.state.confirmed } );
+	},
+
+	onMessageChange( event ) {
+		this.setState( {
+			message: event.target.value
+		} );
+	},
+
+	renderHelpDialog() {
+		const selectedReason = this.state.selectedReason;
+
+		if ( ! selectedReason ) {
+			return;
+		}
+
+		return (
+			<div className="confirm-cancel-domain__help_message">
+				<p>
+					{ selectedReason.helpMessage }
+				</p>
+				{ selectedReason.showTextarea &&
+				<FormTextarea onChange={ this.onMessageChange } rows="5" cols="60" /> }
+			</div>
+		);
+	},
+
+	renderConfirmationCheckbox() {
+		if ( ! this.isValidReasonToCancel() ) {
+			return;
+		}
+
+		return (
+			<div className="confirm-cancel-domain__confirm_container">
+				<FormLabel>
+					<FormCheckbox name="confirm"
+							checked={ this.state.confirmed }
+							onChange={ this.onConfirmationChange } />
+					<span>
+						{ this.translate(
+							'I understand that canceling means that I may {{strong}}lose this domain forever{{/strong}}.', {
+								components: {
+									strong: <strong />
+								}
+							}
+						) }
+					</span>
+				</FormLabel>
+			</div>
+		);
+	},
+
+	renderSubmitButton() {
+		if ( ! this.isValidReasonToCancel() ) {
+			return;
+		}
+
+		if ( this.state.submitting ) {
+			return (
+				<FormButton disabled={ true } >
+					{ this.translate( 'Cancelling Domain…' ) }
+				</FormButton>
+			);
+		}
+
+		const selectedReason = this.state.selectedReason,
+			confirmed = this.state.confirmed;
+
+		if ( selectedReason && 'misspelled' === selectedReason.value ) {
+			return (
+				<FormButton onClick={ this.onSubmit } disabled={ ! confirmed } >
+					{ this.translate( 'Cancel Anyway' ) }
+				</FormButton>
+			);
+		}
+
+		return (
+			<FormButton onClick={ this.onSubmit } disabled={ ! confirmed } >
+				{ this.translate( 'Cancel Domain' ) }
+			</FormButton>
+		);
+	},
+
+	render() {
+		if ( this.isDataLoading() ) {
+			return <ConfirmCancelDomainLoadingPlaceholder
+				purchaseId={ this.props.purchaseId }
+				selectedSite={ this.props.selectedSite } />;
+		}
+
+		const purchase = getPurchase( this.props );
+
+		if ( ! isDomainRegistration( purchase ) ) {
+			return;
+		}
+
+		const domain = getDomainName( purchase ),
+			selectedReason = this.state.selectedReason;
+
+		return (
+			<Main className="confirm-cancel-domain">
+				<HeaderCake onClick={ goToCancelPurchase.bind( null, this.props ) }>
```

Comment:
We have a linting rule for not using Function.prototype.bind in a render() because it creates a new function and forces this part of the tree to re-render. Instead, we can add a class method `this.goToCancelPurchase`, which calls `goToCancelPurchase( this.props )`.